packages/reports/src/Stores/Modules/Statement/Helpers/format-response.ts

```diff
@@ -1,36 +1,42 @@
 import { formatMoney, toTitleCase, toMoment, getMarketInformation, getSymbolDisplayName } from '@deriv/shared';
 import { localize } from '@deriv/translations';
+import { ActiveSymbols, Statement } from '@deriv/api-types';
 
-export const formatStatementTransaction = (transaction, currency, active_symbols = []) => {
+export const formatStatementTransaction = (
+    transaction: NonNullable<Statement['transactions']>[number],
+    currency: string,
+    active_symbols: ActiveSymbols = []
+) => {
+    const { action_type, app_id, contract_id, longcode, purchase_time, withdrawal_details } = transaction;
     const format_string = 'DD MMM YYYY HH:mm:ss';
     const transaction_time = toMoment(transaction.transaction_time).format(format_string);
-    const payout = parseFloat(transaction.payout);
-    const amount = parseFloat(transaction.amount);
-    const balance = parseFloat(transaction.balance_after);
+    const payout = transaction.payout ?? NaN;
+    const amount = transaction.amount ?? NaN;
+    const balance = transaction.balance_after ?? NaN;
```

Comment:
parseFloat was used here to ensure that if the value is undefined or null, it's converted into NaN. transaction.payout can come as a number, null or undefined from API. transaction.amount and transaction.balance_after can come as a number or undefined from API. parseFloat() in TypeScript accepts only a string, that's why I'm using these values directly and fallback to NaN to keep the existing functionality.